Move all handler test to integration tests

w84thesun opened the pull request:
Closes #417.

AlekSi commented on integration/commands_administration_test.go:
```diff
@@ -193,3 +198,132 @@ func TestCommandsAdministrationGetParameter(t *testing.T) {
 }
+
+func TestStatisticsCommands(t *testing.T) {
+	t.Skip("TODO: response deltas investigation needed")
```
There should be an issue URL

w84thesun replied:
Fixed

AlekSi commented on integration/commands_administration_test.go:
```diff
@@ -193,3 +198,132 @@ func TestCommandsAdministrationGetParameter(t *testing.T) {
+
+	for name, tc := range map[string]struct {
+		command  any
+		response map[string]any
```
Unless it is very hard, we should compare `bson.D`s to also check fields order

w84thesun replied:
I used `map[string]any` to skip fields like `gitVersion` or `debug` field which requires me to use `util/version` package.
If it's ok to use it then I can change `response` field to `bson.D`.

w84thesun replied:
I've changed to `bson.D` for now.

AlekSi commented on integration/query_projection_test.go:
```diff
@@ -0,0 +1,122 @@
+	}{
+		"FindProjectionInclusions": {
+			filter: bson.D{{"_id", "document-composite"}},
+			// TODO: fix projection
```
Issue URL

w84thesun replied:
Fixed

AlekSi commented on integration/query_projection_test.go:
```diff
@@ -0,0 +1,122 @@
+			cursor, err := collection.Find(
+				ctx,
+				tc.filter,
+				options.Find().SetProjection(tc.projection),
```
That seems to be short enough to put on one line. One of the `expectred` documents above is longer

w84thesun replied:
Fixed

AlekSi commented on internal/handlers/handler_fuzz_test.go:
```diff
@@ -17,14 +17,55 @@ package handlers
+// Using shared objects helps us spot concurrency bugs.
+// If some test is failing and the log output is confusing, and you are tempted to move setup call to subtest,
+// instead run that single test with `go test -run test/name`.
+func setup(t testing.TB, opts *setupOpts) (context.Context, *pg.Handler, *pgdb.Pool) {
```
I don't think that all should be in that file – that one is about fuzzing

w84thesun replied:
Should I leave that code in `handler_test.go`?

AlekSi replied:
Ah, I see. Yeah, let's leave it there for now

AlekSi reviewed (commented):
Nice catch with a typo in file name!

w84thesun commented:
> Looking into coverage report, I wonder if we deleted some tests without moving them. https://app.codecov.io/gh/FerretDB/FerretDB/compare/523/changes shows lines like that:
> 
> ![image](https://user-images.githubusercontent.com/11512/165352222-b45af2b1-b446-4b27-aa2f-76569450ab16.png)
> 
> Did we forgot tests for negative results?

I've compared tests, and added one missing and two more new ones. The coverage shows those tests which now is rather disabled (stats) or not working correctly (projection).
